crates/weaver_resolver/src/registry.rs

```diff
@@ -680,4 +674,51 @@ mod tests {
 
         Ok(())
     }
+
+    #[test]
+    fn test_api_usage() -> Result<(), Box<dyn Error>> {
```

Comment:
If you use `#[from]` annotation on errors (using ThisError), you can lift errors from other crates into a shared Error type and avoid Box/dyn here.

The alternative is just to implement `From<OtherError> for Error`.

I think we should definitely (eventually) do that for weaver.  Not hugely important for this test.